### PR TITLE
Add option to specify port

### PR DESCRIPTION
**Pull request by pysqz**
fix issue [#157](https://github.com/pynecone-io/pynecone/issues/157)

### CODE REVIEW

File: pynecone/utils.py [hunk 1]
```diff
@@ -533,6 +537,13 @@ def get_num_workers() -> int:
     return (os.cpu_count() or 1) * 2 + 1
 
 
+def get_api_port():
```
Comment (picklelo) [inline pynecone/utils.py:540]: This needs a return type annotation `-> str` as well as a doctoring to make the build happy

File: pynecone/utils.py [hunk 2]
```diff
@@ -533,6 +537,13 @@ def get_num_workers() -> int:
...
+    result = urlparse(get_config().api_url)
+    if result.port is None:
+        return urlparse(constants.API_URL).port
```
Comment (picklelo) [inline pynecone/utils.py:543]: You may need to add an assert here for Pyright
```
port = urlparse(...)
assert port is not None
return port
```

File: pynecone/utils.py [hunk 3]
```diff
@@ -559,6 +571,8 @@ def run_backend_prod(
     """
     num_workers = get_num_workers()
     command = constants.RUN_BACKEND_PROD + [
+        "--bind",
+        "0.0.0.0:%s"%get_api_port(),
```
Comment (picklelo) [inline pynecone/utils.py:575]: Can we use an f-string here for consistency with the rest of the code?

**Review by picklelo — CHANGES_REQUESTED** [review]
Thanks for making this change!

**Comment by Alek99** [comment]
Yes thanks for implementing this feature defiantly needed! 🙂

**Comment by pysqz** [comment]
update the commit, pls review again~ @picklelo @Alek99

**Review by picklelo — APPROVED** [review]
Looks good to me - it just needs to be linted. You can run
```
poetry run black pynecone
```
and it should format the code for you, then we can merge this

**Comment by pysqz** [comment]
> Looks good to me - it just needs to be linted. You can run
> 
> ```
> poetry run black pynecone
> ```
> 
> and it should format the code for you, then we can merge this

done

**Pull request merged**